### ISSUE TITLE
Performance improvement

### ISSUE THREAD

**Issue by pramitchoudhary** [marcotcr/lime#48]
Based on the benchmarks done here
http://stackoverflow.com/questions/20277982/fastest-pairwise-distance-metric-in-python. 
Using  _scipy.spatial.distance.pdist_ vs _sklearn.metrics.pairwise_distances_ could help in improving the computation for computing _euclidean distance_ distance. Need to test it out.

**Comment by marcotcr** [comment]
I don't think it matters one way or the other, sklearn distance is pretty fast already.
The time spent computing distances is always vastly smaller than the time spending with everything else.
Unless you found a case where the above isn't true, it isn't worth the effort.
Thanks : )
